radiometric_normalization/time_stack.py

```diff
@@ -0,0 +1,214 @@
+'''
+Copyright 2015 Planet Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+''' Creates the time stacks for further analysis
+'''
+
+import numpy
+import logging
+
+from osgeo import gdal, gdal_array
+
+
+def generate(image_paths, method='identity', output_path='time_stack.tif'):
+    ''' This takes a list of image paths and creates a time stack image.
+
+    It assumes that all images
+        - contain the same number of bands
+        - the bands are in the same band order
+        - all the images are of the same size
+        - all the images have the same nodata value
+
+    The output file is only supposed to be used internally so options to change
+    the nodata value and the datatype are not exposed.
+    - The output nodata value is 2 ** 15 - 1
+    - The output image data type is uint16
+    - Since the output file is only meant to be used internally, the
+        geographic information isn't carried over to the output file.
+
+    Input:
+        image_paths (list of str): A list of the input file paths
+        method (str): Time stack creation method [identity]
+        output_path (str): A path to write the file to
+    '''
+
+    output_nodata = 2 ** 15 - 1
+    output_datatype = numpy.uint16
+
+    all_bands = _read_in_bands(image_paths)
+    if method is 'identity':
+        output_bands, mask = _mean_with_uniform_weight(
+            all_bands, output_nodata, output_datatype)
+    _write_out_bands(output_bands, mask, output_path, output_nodata)
+
+    return output_path
+
+
+def _read_in_bands(image_paths):
+    ''' Reads in a list of image paths and outputs a list of numpy arrays
+    representing the bands.
+
+    Input:
+        image_paths (list of str): A list of the input file paths
+
+    Output:
+        all_bands (list of list of arrays): A list of each band,
+            each band consisting of a list of images each entry being a
+            masked array of the image data
+    '''
+
+    all_images, nodata = _read_in_gdal_images(image_paths)
+    all_bands = _organise_images_to_bands(all_images, nodata)
+
+    return all_bands
+
+
+def _read_in_gdal_images(image_paths):
```

Comment:
Use gimage to load images. This changes the data structure so _read_in_bands will also change